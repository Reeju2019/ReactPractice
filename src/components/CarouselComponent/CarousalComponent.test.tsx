import { shallow } from "enzyme"
import ReactDOM from "react-dom"
import { CarouselComponent } from "."

const item = {
  items: [
    {
      id: 1,
      img: "https://media.geeksforgeeks.org/wp-content/uploads/20210425122739/2-300x115.png",
      alt: "Image One label",
      header: "First slide label",
      caption: "Nulla vitae elit libero, a pharetra augue mollis interdum.",
    },
    {
      id: 2,
      img: "/assets/images/carousel-2.jfif",
      alt: "Image two label",
      header: "2 slide label",
      caption: "Nulla vitae elit libero, a pharetra augue mollis interdum.",
      button: "button",
      textArea: "textArea",
    },
    {
      id: 3,
      alt: "Image 3 labelll",
      header: "3 slide label",
      caption: "Nulla vitae elit libero, a pharetra augue mollis interdum.",
    },
  ],
}

/**
 * Unit testing for Carousel Component
 */
describe("Carousel Component Testing", () => {
  it("render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <CarouselComponent
        interval={5000}
        nextIcon={
          <span aria-hidden="true" className="carousel-control-next-icon" />
        }
        prevIcon={
          <span aria-hidden="true" className="carousel-control-prev-icon" />
        }
        data={item}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it("should render initial layout", () => {
    const component = shallow(
      <CarouselComponent
        interval={5000}
        nextIcon={
          <span aria-hidden="true" className="carousel-control-next-icon" />
        }
        prevIcon={
          <span aria-hidden="true" className="carousel-control-prev-icon" />
        }
        data={item}
      />
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
