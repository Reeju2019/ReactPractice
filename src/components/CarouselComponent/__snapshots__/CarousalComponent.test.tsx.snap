// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Component Testing should render initial layout 1`] = `
Array [
  <React.Fragment>
    <Row
      style={
        Object {
          "display": "block",
          "padding": 30,
          "width": "100%",
        }
      }
    >
      <Carousel
        activeIndex={0}
        bsPrefix="carousel"
        controls={true}
        defaultActiveIndex={0}
        fade={false}
        indicatorLabels={Array []}
        indicators={true}
        interval={5000}
        keyboard={true}
        nextIcon={
          <span
            aria-hidden="true"
            className="carousel-control-next-icon"
          />
        }
        nextLabel="Next"
        onSelect={[Function]}
        onSlid={[Function]}
        onSlide={[Function]}
        pause="hover"
        prevIcon={
          <span
            aria-hidden="true"
            className="carousel-control-prev-icon"
          />
        }
        prevLabel="Previous"
        slide={true}
        touch={true}
        variant={null}
        wrap={true}
      >
        <CarouselItem
          interval={5000}
        >
          <CarousalItemComponent
            alt="Image One label"
            button={null}
            caption="Nulla vitae elit libero, a pharetra augue mollis interdum."
            defaultInterval={5000}
            header="First slide label"
            img="https://media.geeksforgeeks.org/wp-content/uploads/20210425122739/2-300x115.png"
            textArea={null}
            textAreaHandler={[Function]}
          />
        </CarouselItem>
        <CarouselItem
          interval={5000}
        >
          <CarousalItemComponent
            alt="Image two label"
            button="button"
            caption="Nulla vitae elit libero, a pharetra augue mollis interdum."
            defaultInterval={5000}
            header="2 slide label"
            img="/assets/images/carousel-2.jfif"
            textArea="textArea"
            textAreaHandler={[Function]}
          />
        </CarouselItem>
        <CarouselItem
          interval={5000}
        >
          <CarousalItemComponent
            alt="Image 3 labelll"
            button={null}
            caption="Nulla vitae elit libero, a pharetra augue mollis interdum."
            defaultInterval={5000}
            header="3 slide label"
            img={null}
            textArea={null}
            textAreaHandler={[Function]}
          />
        </CarouselItem>
      </Carousel>
    </Row>
  </React.Fragment>,
]
`;
